qcodes/dataset/measurements.py

```diff
@@ -499,10 +501,13 @@ def __enter__(self) -> DataSaver:
         # next set up the "datasaver"
         if self.experiment is not None:
             self.ds = qc.new_data_set(
-                self.name, self.experiment.exp_id, conn=self.experiment.conn
+                self.name, self.experiment.exp_id, conn=self.experiment.conn,
```

Comment:
missed the cache flag here
```suggestion
                self.name, self.experiment.exp_id,
                conn=self.experiment.conn,
                in_memory_cache=self._in_memory_cache
```